frontend/src/components/recipePage.jsx

```diff
@@ -0,0 +1,122 @@
+import { useState } from 'react';
+import './recipePage.css'
+
+export default function RecipePage( {name, img, desc, ingredients, steps} ) {
+
+    const [newIngredient, setNewIngredient] = useState('');
+    const [newStep, setNewStep] = useState('');
+    
+    const [addNewIngredient, setAddNewIngredient] = useState('');
+    const [addNewStep, setAddNewStep] = useState('');
+
+
+    const addInstruction = async () => {
+        if (newStep) {
+          await fetch(
+            `http://localhost:3001/api/recipe/${name}/instruction`,
+            {
+              method: "PUT",
+              headers: { "Content-Type": "application/json" },
+              body: JSON.stringify({ newStep: newStep })
+            }
+          )
+            .then(setNewStep([...steps, newStep]))
+            .catch(error => console.log("Failed to add step: ", error));
+        }
+    };
+
+    const addIngredient = async () => {
+        if (newIngredient) {
+          await fetch(
+            `http://localhost:3001/api/recipe/${name}/ingredient`,
+            {
+              method: "PUT",
+              headers: { "Content-Type": "application/json" },
+              body: JSON.stringify({ newIngredient: newIngredient })
+            }
+          )
+            .then(setNewStep([...ingredients, newIngredient]))
+            .catch(error => console.log("Failed to add step: ", error));
+        }
+    };
+
+    // const addIngredient = () => {
```

Comment:
I would suggest that you add this function back in some way, as it is responsible for updating the frontend. (Your other function only updates the backend, which is what was asked in M5), but on production applications, we will typically want the changes to be displayed on both the frontend and the backend!